js/whitebit.js

```diff
@@ -974,64 +974,65 @@ module.exports = class whitebit extends Exchange {
-                // it's a stop-market-order
+            if (isLimitOrder) {
+                // stop limit order
+                method = 'v4PrivatePostOrderStopLimit';
+                request['price'] = this.priceToPrecision (symbol, price);
+            } else {
+                // stop market order
                 method = 'v4PrivatePostOrderStopMarket';
             }
         } else {
-            if (type === 'market') {
-                // it's a regular market order
-                method = 'v4PrivatePostOrderMarket';
-            }
-            if (type === 'limit') {
-                // it's a regular limit order
+            if (isLimitOrder) {
+                // limit order
                 method = 'v4PrivatePostOrderNew';
+                request['price'] = this.priceToPrecision (symbol, price);
+            } else {
+                // market order
+                method = 'v4PrivatePostOrderMarket';
             }
         }
-        // aggregate common assignments regardless stop or not
-        if (type === 'limit' || type === 'stopLimit') {
+        if (isMarketOrder && side === 'buy' && !fok) {
             if (price === undefined) {
-                throw new ArgumentsRequired (this.id + ' createOrder() requires a price argument for a stopLimit order');
-            }
-            const convertedPrice = this.priceToPrecision (symbol, price);
-            request['price'] = convertedPrice;
-        }
-        if (type === 'market' || type === 'stopMarket') {
-            if (side === 'buy') {
-                let cost = this.safeNumber (params, 'cost');
-                const createMarketBuyOrderRequiresPrice = this.safeValue (this.options, 'createMarketBuyOrderRequiresPrice', true);
-                if (createMarketBuyOrderRequiresPrice) {
-                    if (price !== undefined) {
-                        if (cost === undefined) {
-                            cost = amount * price;
-                        }
-                    } else if (cost === undefined) {
-                        throw new InvalidOrder (this.id + " createOrder() requires the price argument for market buy orders to calculate total order cost (amount to spend), where cost = amount * price. Supply a price argument to createOrder() call if you want the cost to be calculated for you from price and amount, or, alternatively, add .options['createMarketBuyOrderRequiresPrice'] = false and supply the total cost value in the 'amount' argument or in the 'cost' extra parameter (the exchange-specific behaviour)");
-                    }
-                } else {
-                    cost = (cost === undefined) ? amount : cost;
-                }
-                request['amount'] = this.costToPrecision (symbol, cost);
+                throw new InvalidOrder (this.id + ' createOrder () requires the price argument with market buy orders to calculate total order cost (amount to spend), where cost = amount * price.');
```

Comment:
Shouldn't we be using the `createMarketBuyOrderRequiresPrice` option here?

Reply:
There no option, to `not` supply it as a cost, I though that the `createMarketBuyOrderRequiresPrice` is for when you have the choice of; supplying a price and amount separately, or a together in a cost as an amount to spend. Or am I wrong and its simply an informative `option` rather than a functional one?

Reply:
@DoctorSlimm Not totally sure I understood the question 😅 

Exchanges like whitebit, require the `cost/quote currency` for market buy order right? We don't have a choice there, we have to always send the cost as the amount instead of the base currency.

So if the user provides only the base currency, let's say 0.001BTC  (assuming we're trading BTC/USDT) we don't have the means to estimate the cost so we have to require the price as well and the user would call `createOrder` like this:

```Javascript
order = exchange.createOrder("BTC/USDT", "market", "buy", 0.001, 10000) // cost here is 10USDT
```

However, some users are aware of this "behavior" and prefer to use the `cost` value directly as the amount, instead of the hassle of having to provide a price too, so they might do something like this:

```Javascript
order = exchange.createOrder("BTC/USDT", "market", "buy", 10) // <-- passing the `cost` as the amount
```
The option `createMarketBuyOrderRequiresPrice` controls precisely that and the user has to explicitly set it as `false` if he wants to be able to provide only the amount (which is the cost in this case) instead of the amount in base currency + price. Without this option, if we simply accepted the amount as the cost,  the user could unintentionally do something like this:

```Javascript
order = exchange.createOrder("SHIB/USDT", "market", "buy", 1000000) 
```
⬆ user thinks it's buying 10000000 SHIB (12 USDT) but in reality bought 1 million dollars worth of SHIB.

Is it a bit more clear now?

Reply:
Oh right I get it thanks, I misunderstood the usage of the 
`createMarketBuyOrderRequiresPrice` option.